### PR TITLE
Review jdk instr

### PR DESCRIPTION
**Pull request by vext01**
The JDK instrumentation is all but finished in master, but I noticed a possible issue. It's not clear of GC collector names are unique (there may be two of the same name). So here I am switching from using a mapping (collector names -> info) to a sequence, thus allowing duplicate collector names.

Also improved comments.

OK?

### CODE REVIEW

File: iterations_runners/iterations_runner.java [hunk 1]
```diff
@@ -29,9 +29,18 @@ private String getCompJSON() {
...
+    /**
+     * Get GC info for the specified collector bean.
+     *
+     * @param gcBean Collector in question
+     * @return string of the form [name, [pools, time, count]]
+     */
     private String getGCJSON(GarbageCollectorMXBean gcBean) {
-        String json = "[";
+        /*
+         * Note, resisted the temptation to use a mapping from name
```
Comment (snim2) [inline iterations_runners/iterations_runner.java]: Could also be a Javadoc comment

**Comment by snim2** [comment]
This looks fine to me, I'll trust that you've tested it!

File: iterations_runners/iterations_runner.java (hunk 1 above)
Comment (ltratt) [inline iterations_runners/iterations_runner.java]: Since this is inside a function, surely this would be neater with the `//` syntax?

**Comment by ltratt** [comment]
How big does the JSON string get? Won't `+=` have non-linear performance? I think Java has something like StringBuilder which would be close to linear?

**Comment by vext01** [comment]
It doesn't get big, but perhaps we should use a stringbuilder anyway.

**Comment by ltratt** [comment]
The stringbuider thingy looks good to me. Thanks. Will let Sarah complete the PR.

**Comment by vext01** [comment]
@snim2 I have inspected the outcome and there are a couple of simple unit tests that just ensure the JSON is passed through into the result file.

OK to squash?

File: iterations_runners/iterations_runner.java [hunk 2]
```diff
@@ -24,63 +24,69 @@ public KrunJDKInstrumentation() {
...
-    private String getCompJSON() {
-        return "" + this.compBean.getTotalCompilationTime();
+    private void getCompJSON(StringBuilder sb) {
+        sb.append(this.compBean.getTotalCompilationTime());
     }
 
 
     /**
-     * Get GC info for the specified collector bean.
+     * Write GC info for the specified collector bean to a StringBuilderc.
```
Comment (vext01) [inline iterations_runners/iterations_runner.java]: typo

**Comment by snim2** [comment]
Cool, squash away

**Comment by vext01** [comment]
Just wait for travis...

**Comment by snim2** [comment]
Travis is good...

**Comment by vext01** [comment]
It's squashed Jim, but not as we know it.

**Pull request merged**